### PR TITLE
Split log category's into individual files

### LINKED ISSUE #602: output.log is never reset and can grow indefinitely
### Summary
output.log can grow indefinitely. We should probably put some kind of cap on it's size, and have it reset every X number of sessions.

My own log exceeded 700MB before I deleted. Admittedly, I was doing a lot of movement testing, with a lot of output over dozens and dozens of sessions, but still, that's huge.

### PR DESCRIPTION
**Pull request by miles200**
- Add ability to 'combine' or 'split' log files 
- If splitting: For each QLoggingCategory a separate, dated log file is created. 
- If combining: A single 'all', dated log file is created
- Above 2 items controlled using a new 'combine_logs' setting in settings.cfg under 'Logging'. Also added to settings_template.cfg
- Add functionality to SEGSAdmin to control this setting.
- Add functionality to SEGSAdmin to 'Purge Logs'. This will delete any log files with a date < today. Convenience function to allow easy clean up of old log files.
- Some SEGSAdmin UI tidy ups, some issues had arisen following previous updates
- Code indentation fix in SettingsDialog.cpp

### Issues closed
Closes #602 

## SEGSAdmin Testing

image

### CODE REVIEW

File: Projects/CoX/Utilities/SEGSAdmin/SettingsDialog.cpp [hunk 1]
```diff
@@ -97,93 +100,98 @@ void SettingsDialog::read_config_file(QString filePath)
...
-        config_file.endGroup();
-        config_file.beginGroup("CharacterDatabase");
-        QString char_db_driver = config_file.value("db_driver","").toString();
-        QString char_db_host = config_file.value("db_host","").toString();
-        int char_db_port = config_file.value("db_port","").toInt();
-        int char_index = ui->char_dbdriver->findText(char_db_driver);
-        ui->char_dbdriver->setCurrentIndex(char_index);
-        ui->char_dbhost->setText(char_db_host);
-        ui->char_dbport->setValue(char_db_port);
+    QString acc_db_driver = config_file.value("db_driver","").toString();
```
Comment (broxen) [inline Projects/CoX/Utilities/SEGSAdmin/SettingsDialog.cpp]: I'd actually prefer to have these indented, as it makes it clear what's part of each config group

Comment (miles200) [reply Projects/CoX/Utilities/SEGSAdmin/SettingsDialog.cpp]: But it looks so nasty :) Can we add a line break between each group?

Comment (broxen) [reply Projects/CoX/Utilities/SEGSAdmin/SettingsDialog.cpp]: Oh sure. I'm good with that.

**Comment by miles200** [comment]
Testing requirements added for testers

File: Projects/CoX/Utilities/SEGSAdmin/SettingsDialog.cpp [hunk 2]
```diff
@@ -413,4 +442,72 @@ void SettingsDialog::send_maps_dir_config_check()
...
+    QStringList logs_to_delete;
+
+    for (QString &file : existing_logs)
+    {
+        QStringList parts = file.split("_");
+        if (!parts.isEmpty())
+        {
+            QString date = parts.value(parts.length() - 1);
+            date.replace(".log", "");
+            int compare = QString::compare(date, todays_date, Qt::CaseInsensitive);
```
Comment (dracc) [inline Projects/CoX/Utilities/SEGSAdmin/SettingsDialog.cpp]: Did this comparison receive any testing? What happens when March 31st turns into April 1st? Or April 30th for that matter?

**Pull request merged**